### PR TITLE
[UNR-4415] Add a dedicated authority component set for Server Worker Entities

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description
Part of UNR-4415 is allowing to send Unordered Reliable CrossServer RPC. We chose to use the ServerWorker entity as a sender for this operation.
This creates the situation where the server worker entity would need to have authority over some components traditionally associated to Actors, namely the cross server sender.
It also means that it needs authority over it. Trying to add the ServerAuthority component set to the authority delegation resulted in deployments not starting anymore.
What it looks like is that overlapping component sets are not supported in the authority delegation component, which make sense, because they could be set to different partitions.
A dedicated authority component set was added specifically for server worker entities.

#### Release note
Internal change

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 1]
```diff
@@ -55,13 +55,14 @@ void ServerWorkerEntityCreator::CreateWorkerEntity()
...
 	DelegationMap.Add(SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID, EntityId);
-	Components.Add(AuthorityDelegation(DelegationMap).CreateComponentData());
+	DelegationMap.Add(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, EntityId);
 
 	if (Settings->CrossServerRPCImplementation == ECrossServerRPCImplementation::RoutingWorker)
 	{
 		Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::CROSSSERVER_SENDER_ENDPOINT_COMPONENT_ID));
 		Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::CROSSSERVER_SENDER_ACK_ENDPOINT_COMPONENT_ID));
 	}
+	Components.Add(AuthorityDelegation(DelegationMap).CreateComponentData());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:65]: Any reason for moving this to here?

**Comment by m-samiec** [comment]
Couldn't we also have added the specific components needed for sending rpcs to the `KnownEntityAuthComponentSet` ?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp (hunk 1 above)
Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:65]: Yes, I will later add a new entry in the if() to delegate authority of some components to the routing worker.

**Comment by ImprobableNic** [comment]
We could, but I would rather move towards disjoints component set, since we know we can't use overlapping sets together.

**Comment by m-samiec** [comment]
Problem with disjoints is that they can't be updated atomically though

**Comment by ImprobableNic** [comment]
That is true, I do not think it is a problem for this particular one, but I'll change my approach.

File: SpatialGDK/Extras/schema/server_worker.schema [hunk 2]
```diff
@@ -21,3 +22,14 @@ component ServerWorker {
...
 }
+
+component_set ServerWorkerAuthComponentSet {
+  id = 9908;
+  components = [
+	improbable.Position,
+	improbable.Metadata,
+	improbable.Interest,
+	unreal.ServerWorker,
+    unreal.generated.UnrealCrossServerSenderRPCs
```
Comment (m-samiec) [inline SpatialGDK/Extras/schema/server_worker.schema:33]: spacing is off here

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 3]
```diff
@@ -54,14 +54,16 @@ void ServerWorkerEntityCreator::CreateWorkerEntity()
...
 	AuthorityDelegationMap DelegationMap;
-	DelegationMap.Add(SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID, EntityId);
-	Components.Add(AuthorityDelegation(DelegationMap).CreateComponentData());
+	DelegationMap.Add(SpatialConstants::SERVER_WORKER_ENTITY_AUTH_COMPONENT_SET_ID, EntityId);
 
 	if (Settings->CrossServerRPCImplementation == ECrossServerRPCImplementation::RoutingWorker)
 	{
 		Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::CROSSSERVER_SENDER_ENDPOINT_COMPONENT_ID));
 		Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::CROSSSERVER_SENDER_ACK_ENDPOINT_COMPONENT_ID));
+		Components.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::ROUTINGWORKER_TAG_COMPONENT_ID));
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:63]: This doesn't look necessary for this pr - just a driveby?

**Review by m-samiec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp [hunk 4]
```diff
@@ -126,7 +126,7 @@ void UGlobalStateManager::TrySendWorkerReadyToBeginPlay()
 		StaticComponentView->HasComponent(GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID);
 	const bool bWorkerEntityReady =
 		NetDriver->WorkerEntityId != SpatialConstants::INVALID_ENTITY_ID
-		&& StaticComponentView->HasAuthority(NetDriver->WorkerEntityId, SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID);
+		&& StaticComponentView->HasAuthority(NetDriver->WorkerEntityId, SpatialConstants::SERVER_WORKER_ENTITY_AUTH_COMPONENT_SET_ID);
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp:129]: probably needs updating to use either the main view or a relevant subview (well known entity system potentially?)

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 5]
```diff
@@ -1100,12 +1100,25 @@ SCHEMA_GENERATOR_TEST(GIVEN_actor_class_WHEN_generating_schema_THEN_expected_com
...
 
-	TestTrue("Found spatial well known components", SchemaDatabase->ComponentSetIdToComponentIds.Contains(50));
-	if (SchemaDatabase->ComponentSetIdToComponentIds.Contains(50))
+	TestTrue("Found spatial well known components",
+			 SchemaDatabase->ComponentSetIdToComponentIds.Contains(SpatialConstants::SPATIALOS_WELLKNOWN_COMPONENTSET_ID));
+	if (SchemaDatabase->ComponentSetIdToComponentIds.Contains(SpatialConstants::SPATIALOS_WELLKNOWN_COMPONENTSET_ID))
 	{
-		TestTrue("Spatial well know component is not empty", SchemaDatabase->ComponentSetIdToComponentIds[50].ComponentIDs.Num() > 0);
+		TestTrue(
+			"Spatial well know component is not empty",
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:1110]:
```suggestion
			"Spatial well known component is not empty",
```

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:1110]: and below

**Review by jacquese — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp (hunk 3 above)
Comment (ImprobableNic) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp:63]: True, I have overlapping workspace.
